gadopt/utility.py

```diff
@@ -475,3 +477,56 @@ def node_coordinates(function):
     return [
         Function(func_space).interpolate(coords).dat.data for coords in mesh_coords
     ]
+
+
+def assign_1d_profile(function: Function, one_d_filename: str, cartesian: bool, r1d=None):
+    """
+    Assign a one-dimensional profile to a Function `q` from a file.
+
+    The function reads a one-dimensional radial viscosity profile together with the
+    radius input for the profile, from a file, broadcasts the read array to all
+    processes, and then interpolates this array onto the function space of `q`.
+
+    Args:
+        q (Function): The function onto which the 1D profile will be assigned.
+        one_d_filename (str): The path to the file containing the 1D radial viscosity profile.
+        cartesian (bool): Is the upward direction along z/y(cartesian=True) or radial (cartesian=False)
+
+    Returns:
+        None: It directly assigns the input function `function`.
+
+    Note:
+        - Note the cartesian flag
+        - This is designed to read a file with one process and distribute in parallel with MPI.
+        - The input file should contain an array of radius and an array of viscosity values, seperated by comma.
+        - Make sure radius in the file is consistent with the mesh.
```

Comment:
Ah! What I mean hear is that because interp1d (now np.interp )does not raise an error for out of bound values, one should be careful about the dimensions. `LayerAveraging` works with whatever value comes out of `interp`